### PR TITLE
added new models and routes for friends table and added functionality to events 

### PR DESCRIPTION
**Pull request by Cody-Hayes97**
# Description
added helper functions for the friends table. routes are currently in progress

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X ] This change requires a documentation update

## Change Status

- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [X ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X ] There are no merge conflicts

### CODE REVIEW

File: components/friends/friends-model.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+function updateFriend(id, updates) {
+  return db("friends").where({ id }).update(updates);
+}
+
+function removeFriend(id) {
+  return db("friends").where("id", id).del();
+}
+
+function getById(id) {
```
Comment (agyin3) [inline components/friends/friends-model.js:27]: Should we do a join with the `users` table and return both users' information?

**Comment by agyin3** [comment]
Think it might be a good idea to create a helper function that will return all of a user's friends. Maybe a `getFriendsById`

File: components/friends/friends-model.js (hunk 1 above)
Comment (Cody-Hayes97) [reply components/friends/friends-model.js:27]: good idea. Ill implement that

**Review by Katrina-Dierking — APPROVED** [review]
All endpoints tested and confirmed. Great job working through the merge conflicts as well.

**Comment by agyin3** [comment]
think we should also return business info when calling `findBy`

**Comment by agyin3** [comment]
Changes look good for `findById` and `findBy` helpers

**Comment by agyin3** [comment]
I tested on my end and looks great. Thanks for fixing the model. Restricted middleware is also working as wanted

File: .gitignore [hunk 2]
```diff
@@ -115,4 +115,6 @@ dist
 *.sql
 
 # VSCode directories
-.vscode/
\ No newline at end of file
+.vscode/
+
```
Comment (tinomen) [inline .gitignore:119]: Oh, I thought we were going to share the vscode config. Let talk as a team

File: index.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
-const server = require('./server.js');
+const server = require("./server.js");
 
 const PORT = process.env.PORT || 4000;
 
 server.listen(PORT, () => {
-    console.log(`Listening on port ${PORT}...`);
-});
\ No newline at end of file
+  console.log(`Listening on port ${PORT}...`);
```
Comment (tinomen) [inline index.js:6]: Lets get rid of this log

**Pull request merged**